src/client.c

```diff
@@ -476,6 +478,22 @@ bool set_pool(PgSocket *client, const char *dbname, const char *username, const
 		}
 	} else {
 		client->login_user_credentials = find_global_credentials(username);
+
+		/* Check client_connection count limit */
+		if (client->login_user_credentials) {
+			client->login_user_credentials->global_user->client_connection_count++;
```

Comment:
Let's add a comment here for clarity.
```suggestion
			/* increment count now, so that we can decrement it safely in disconnect_client if limit was reached */
			client->login_user_credentials->global_user->client_connection_count++;
```